Sources/Core/CharcoalViewController.swift

```diff
@@ -136,9 +141,24 @@ extension CharcoalViewController: FilterViewControllerDelegate {
     }
 }
 
+extension CharcoalViewController: FilterSelectionStoreDelegate {
+    func filterSelectionStoreDidChange(_ selectionStore: FilterSelectionStore) {
+        selectionHasChanged = true
```

Comment:
How about:
```swift
if topViewController === rootFilterViewController {
    selectionHasChanged = false
    filterDelegate?.charcoalViewController(self, didChangeSelection: selectionStore.queryItems(for: filter.rootFilter))
} else {
    selectionHasChanged = true
}
```

The you don't need `rootFilterViewControllerDidChangeSelection` anymore and rely only on observing changes in `FilterSelectionStore`.

Reply:
Good idea! :D